AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m

```diff
@@ -228,9 +253,12 @@ - (void)deleteLogsFromDBWithColumnValues:(NSArray *)columnValues columnName:(NSS
   // Build up delete query.
   char surroundingChar = ([[columnValues firstObject] isKindOfClass:[NSString class]]) ? '\'' : '\0';
   NSString *valuesSeparation = [NSString stringWithFormat:@"%c, %c", surroundingChar, surroundingChar];
+  NSString *whereCondition = [NSString stringWithFormat:@"\"%@\" IN (%c%@%c)", columnName, surroundingChar,
+                              [columnValues componentsJoinedByString:valuesSeparation], surroundingChar];
+  NSArray<NSArray *> *logEntries = [self logsWithCondition:whereCondition];
+  [self deleteTargetTokensFromKeychain:logEntries];
```

Comment:
Erg, that's a lot to do on the delete: get the log, iterate to get the tokens, delete the tokens from the keychain, delete the logs from the DB. I am trying to think of another technic:
Look like there is a basic encryption mechanism providing by [the iOS framework](https://developer.apple.com/library/content/documentation/Security/Conceptual/cryptoservices/GeneralPurposeCrypto/GeneralPurposeCrypto.html#//apple_ref/doc/uid/TP40011172-CH9-SW14) or if you find any other vanilla way to encrypt data as soon as it's secured enough. We should create a key for this encryption logic, store/get that key in the keychain while the DB is initializing and then use it to encrypt/decrypt directly from the DB. That way there is no need to also manage tokens in the keychain.